packages/avatars/src/styled/utility.ts

```diff
@@ -16,40 +16,40 @@ const [xxs, xs, s, m, l] = SIZE;
 export const TRANSITION_DURATION = 0.25;
 
 export interface IStyledStatusIndicatorProps extends ThemeProps<DefaultTheme> {
-  readonly size?: IAvatarProps['size'];
-  readonly type?: IAvatarProps['status'] | 'active';
+  readonly $size?: IAvatarProps['size'];
+  readonly $type?: IAvatarProps['status'] | 'active';
 }
 
 export function getStatusColor(
-  type?: IStyledStatusIndicatorProps['type'],
-  theme?: IStyledStatusIndicatorProps['theme']
+  theme: IStyledStatusIndicatorProps['theme'],
+  type?: IStyledStatusIndicatorProps['$type']
 ): string {
   switch (type) {
```

Comment:
Should this `switch` lookup be replaced with a mapping object?

Reply:
Do you mean something more like this?

```
export function getStatusColor(
  theme: IStyledStatusIndicatorProps['theme'],
  type?: IStyledStatusIndicatorProps['$type']
): string {
  return (
    {
      active: getColor({ hue: 'crimson', light: { shade: 700 }, dark: { shade: 600 }, theme }),
      available: getColor({ hue: 'mint', light: { shade: 500 }, dark: { shade: 400 }, theme }),
      away: getColor({ hue: 'orange', light: { shade: 500 }, dark: { shade: 400 }, theme }),
      transfers: getColor({ hue: 'azure', light: { shade: 500 }, dark: { shade: 400 }, theme }),
      offline: getColor({ hue: 'grey', light: { shade: 500 }, dark: { shade: 400 }, theme })
    }[type as string] || 'transparent'
  );
}
```